core/eolearn/core/eoexecution.py

```diff
@@ -169,6 +171,7 @@ def _execute_workflow(cls, process_args):
         stats[cls.STATS_END_TIME] = dt.datetime.now()
 
         if log_path:
+            logger.info(msg='Pipeline failed.' if cls.STATS_ERROR in stats else 'Pipeline finished.')
             handler.close()
             logger.removeHandler(handler)
```

Comment:
It might be good to put this code under try-catch statement as well. This is for the case if adding a log handler would fail. Current code would catch it but then removing a non-existing log handler would fail as well. Maybe we could even split this logic into new methods and make this main method more readable:

```Python
workflow, input_args, log_path, return_results = process_args

logger, handler = self._try_add_logging(log_path)

stats = {cls.STATS_START_TIME: dt.datetime.now()}
try:
    results = workflow.execute(input_args, monitor=True)

    if return_results:
        stats[cls.RESULTS] = results

except BaseException:
    stats[cls.STATS_ERROR] = traceback.format_exc()

stats[cls.STATS_END_TIME] = dt.datetime.now()

self._try_remove_logging(log_path, logger, handler)

return stats
```
where content of `_try_add_logging` and `_try_remove_logging` would be approximately in a form
```Python
if log_path:
    try:
        ...
    except BaseException:
        pass
```